### PR TITLE
Better selection support

### PR DESCRIPTION
**Pull request by gaborbernat**
Signed-off-by: Bernát Gábor <gaborjbernat@gmail.com>

### CODE REVIEW

File: src/tox/session/env_select.py [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+        parser.add_argument("-e", dest="env", help="environment to run", default=default, type=CliEnv)
+        return
+
+    if not group_only:
+        group = parser.add_argument_group("select target environment(s)")
+        excl = group.add_mutually_exclusive_group(required=False)
+        help_msg = "enumerate (ALL -> all environments, not set -> use <env_list> from config)"
+        excl.add_argument("-e", dest="env", help=help_msg, default=default, type=CliEnv)
+        # excl.add_argument("-m", dest="labels", metavar="label", help="labels to run", default=[], type=str, nargs="+")
+        # excl.add_argument("-f", dest="factors", help="factors to run", default=[], type=str, nargs="+")
```
Comment (jugmac00) [inline src/tox/session/env_select.py:72]: leftovers?

**Review by jugmac00 — APPROVED** [review]
Looks great! :rocket: 

This should fix #238.

I have one minor inline comment.

Also, if possible, please add some documentation.

File: src/tox/session/env_select.py (hunk 1 above)
Comment (gaborbernat) [reply src/tox/session/env_select.py:72]: More like to do.  The current changes just refactor the world to make factor and tag base selection possible.  I should have made this PR a draft, sorry 😁

**Comment by ericchansen** [comment]
I'm trying to learn how to use this feature, and I'm struggling. Is there any particular docs I should focus on? Or perhaps certain tests that demonstrate the feature well?

**Comment by gaborbernat** [comment]
What you looking to achieve?

**Comment by jugmac00** [comment]
> I'm trying to learn how to use this feature, and I'm struggling. Is there any particular docs I should focus on? Or perhaps certain tests that demonstrate the feature well?

Does this link answer your question? https://tox.wiki/en/rewrite/config.html#labels

If not, please create an issue.

**Comment by ericchansen** [comment]
@jugmac00, I read that before making this comment, but thank you!

@gaborbernat I want to create a new env that runs in almost the same way as the default env except it has more extras. I'll try to give a minimal example.

```
[tox]
envlist = py36, py37, py38, py39
minversion = 3.15

[testenv]
description = Invoke pytest to run automated tests
commands =
    pytest \
        --cov \
        --cov-append \
        --junitxml {toxworkdir}/junit.{envname}.xml \
        src/tests
        {posargs}
extras =
    testing
passenv =
    HOME
setenv =
    COVERAGE_FILE = {env:COVERAGE_FILE:{toxworkdir}/.coverage.{envname}}
    TOXINIDIR = {toxinidir}

[testenv:gitlab]
extras =
    gitlab
    testing
```

I want the `gitlab` env to do the same thing as the default env (run the tests on py36, py37, etc.), except also install a few more additional dependencies.

**Pull request merged**